src/libraries/System.Private.CoreLib/src/System/MathF.cs

```diff
@@ -88,24 +90,48 @@ public static float BitIncrement(float x)
             return BitConverter.Int32BitsToSingle(bits);
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static unsafe float CopySign(float x, float y)
         {
-            // This method is required to work for all inputs,
-            // including NaN, so we operate on the raw bits.
+            const int signMask = unchecked((int)0b_1000_0000__0000_0000__0000_0000__0000_0000);
```

Comment:
Same for the 32bit version. And also, here is not consistent with the `1L<<63` definition.